apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SqlResultStream.cs

```diff
@@ -256,9 +247,13 @@ internal bool CanRetry(RpcException exception)
             /// </summary>
             internal async Task RecordErrorAndWaitAsync(RpcException exception, CancellationToken cancellationToken)
             {
-                _consecutiveErrors++;
                 TimeSpan? delayFromException = GetRetryDelay(exception);
-                await _scheduler.Delay(_retrySettings.BackoffJitter.GetDelay(delayFromException ?? _retrySettingsBackoffs.Current), cancellationToken).ConfigureAwait(false);
+                TimeSpan delay = _retrySettings.BackoffJitter.GetDelay(delayFromException ?? _retrySettingsBackoffs.Current);
+                if (_clock.GetCurrentDateTimeUtc() + delay >= _currentDeadline)
+                {
+                    throw new RpcException(new Grpc.Core.Status(StatusCode.DeadlineExceeded, "Streaming call exceeded timeout"));
```

Comment:
nit: should this be a SpannerException? It's our combined operation that has timed out, not a specific RPC? If a specific RPC times out, then CanRetry should be false and the RpcException for the deadline should be thrown.
Not sure though if it makes sense to make that distinction.

Reply:
I think it's better to throw an `RpcException` here to keep it as similar to other `DeadlineExceeded` exceptions that might occur. This `RpcException` is [automatically wrapped](https://github.com/googleapis/google-cloud-dotnet/blob/34b71e8b87efdbc3043906d9b2dd7d1b8e9bb43d/apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerDataReader.cs#L309) in a `SpannerException`, and anyone catching that exception and then unwrapping the underlying `RpcException` would then get the same behavior regardless whether the exception is thrown here or whether it actually occurs while the RPC is being executed (which in this case could also happen if for example the backoff time is less than the remaining time to the deadline, but the deadline is reached while retrying the RPC after the backoff).

Reply:
OK